examples/tests/doublefree.py

```diff
@@ -41,18 +39,12 @@ def test(self):
         Execute 'doublefree'.
         """
         cmd = os.path.join(self.srcdir, 'doublefree')
-        cmd_result = process.run(cmd, ignore_status=True)
+        cmd_result = process.run(cmd, ignore_status=True,
+                                 env={'MALLOC_CHECK_': '1'})
         self.log.info(cmd_result)
-        expected_exit_status = -signal.SIGABRT
-        output = cmd_result.stdout + cmd_result.stderr
-        self.assertEqual(cmd_result.exit_status, expected_exit_status)
-        if sys.platform.startswith('darwin'):
-            pattern = 'pointer being freed was not allocated'
-        else:
-            pattern = 'double free or corruption'
-        self.assertTrue(pattern in output,
-                        msg='Could not find pattern %s in output %s' %
-                            (pattern, output))
+        pattern = 'free(): invalid pointer'
```

Comment:
Shouldn't the original pattern on darwin be kept?  We can merge this fix if we don't change that and if we verify it fixes the issue on non-darwin platforms.

BTW, I did some research and it looks like only glibc based systems would be affected by the `MALLOC_CHECK_` environment variable, which again should make a non-darwin only fix OK.

Reply:
ok, let's keep it then.

Reply:
BTW, apart from keeping the darwin stuff, I've an ack here based on this test:
```
2017-11-30 08:52:36,850 test             L0274 DEBUG| DATA (filename=doublefree.c) => /home/cleber/src/avocado/examples/tests/doublefree.py.data/doublefree.c (found at file source dir)
2017-11-30 08:52:36,850 process          L0509 INFO | Running 'make doublefree'
2017-11-30 08:52:36,854 process          L0362 DEBUG| [stdout] cc -g -O0    doublefree.c   -o doublefree
2017-11-30 08:52:36,886 process          L0585 INFO | Command 'make doublefree' finished with 0 after 0.0339348316193s
2017-11-30 08:52:36,898 process          L0509 INFO | Running '/var/tmp/avocado_p2MPgl/1-examples_tests_doublefree.py_DoubleFreeTest.test/src/doublefree'
2017-11-30 08:52:36,900 process          L0585 INFO | Command '/var/tmp/avocado_p2MPgl/1-examples_tests_doublefree.py_DoubleFreeTest.test/src/doublefree' finished with 0 after 0.000382900238037s
2017-11-30 08:52:36,900 process          L0362 DEBUG| [stderr] *** Error in `/var/tmp/avocado_p2MPgl/1-examples_tests_doublefree.py_DoubleFreeTest.test/src/doublefree': free(): invalid pointer: 0x0000000001a14010 ***
2017-11-30 08:52:36,912 doublefree       L0044 INFO | Command: /var/tmp/avocado_p2MPgl/1-examples_tests_doublefree.py_DoubleFreeTest.test/src/doublefree
Exit status: 0
Duration: 0.000382900238037
Stdout:

Stderr:
*** Error in `/var/tmp/avocado_p2MPgl/1-examples_tests_doublefree.py_DoubleFreeTest.test/src/doublefree': free(): invalid pointer: 0x0000000001a14010 ***

PID:
4464

2017-11-30 08:52:36,913 test             L0278 DEBUG| DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
2017-11-30 08:52:36,913 test             L0278 DEBUG| DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
2017-11-30 08:52:36,913 test             L0976 INFO | PASS 1-examples/tests/doublefree.py:DoubleFreeTest.test
2017-11-30 08:52:36,913 test             L0959 INFO | 
2017-11-30 08:52:36,942 job              L0478 INFO | Test results available in /home/cleber/avocado/job-results/job-2017-11-30T08.52-b5e6be
```